api/model/src/main/java/org/projectnessie/model/NessieUserInfo.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.model;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.immutables.value.Value;
+
+@Schema(
+    type = SchemaType.OBJECT,
+    title = "NessieUserInfo",
+    description = "Information about the current user.")
+@Value.Immutable
+@JsonSerialize(as = ImmutableNessieUserInfo.class)
+@JsonDeserialize(as = ImmutableNessieUserInfo.class)
+public interface NessieUserInfo {
+  @Value.Parameter(order = 1)
+  boolean anonymous();
+
+  @Value.Parameter(order = 2)
+  @Nullable
+  @jakarta.annotation.Nullable
+  String name();
+
+  @Value.Parameter(order = 3)
+  List<String> roles();
```

Comment:
Do we have to expose roles? In general, I tend to think the less auth info we expose, the better.

Reply:
How does this expose "more" than you could get yourself from an OAuth2 server anyway?

Reply:
The Nessie server may have more privileges for introspecting users than the actual user.

Reply:
In other words, this could (potentially) be a way for the (malicious) user to discover something about the authenticated identity that is not meant to be available by other means.... this is potential, of course, but if we do not have to expose this information, I think we ought not to expose it.

Reply:
But again: how is that different from any other way to authenticate and ask the oauth2 server? Or how Linux' `id` command shows all the groups? Nothing says that the server _has to_ return all roles - it _can_.